drivers/can/can_mspm0_canfd.c

```diff
@@ -0,0 +1,276 @@
+	uint32_t ti_canfd_base;
+	const struct mspm0_sys_clock *clock_subsys;
+
+	const DL_MCAN_ClockConfig clock_cfg;
+	mm_reg_t mcan_base;
+	mem_addr_t mram;
+
+	void (*irq_cfg_func)(void);
+	const struct pinctrl_dev_config *pinctrl;
+};
+
+static int can_mspm0_canfd_read_reg(const struct device *dev, uint16_t reg, uint32_t *val)
+{
+	const struct can_mcan_config *mcan_config = dev->config;
+	const struct can_mspm0_canfd_config *config = mcan_config->custom;
+
+	return can_mcan_sys_read_reg(config->mcan_base, reg, val);
+}
+
+static int can_mspm0_canfd_write_reg(const struct device *dev, uint16_t reg, uint32_t val)
+{
+	const struct can_mcan_config *mcan_config = dev->config;
+	const struct can_mspm0_canfd_config *config = mcan_config->custom;
+
+	return can_mcan_sys_write_reg(config->mcan_base, reg, val);
+}
+
+static int can_mspm0_canfd_read_mram(const struct device *dev, uint16_t offset, void *dst,
+				     size_t len)
+{
+	const struct can_mcan_config *mcan_config = dev->config;
+	const struct can_mspm0_canfd_config *config = mcan_config->custom;
+
+	return can_mcan_sys_read_mram(config->mram, offset, dst, len);
+}
+
+static int can_mspm0_canfd_write_mram(const struct device *dev, uint16_t offset, const void *src,
+				      size_t len)
+{
+	const struct can_mcan_config *mcan_config = dev->config;
+	const struct can_mspm0_canfd_config *config = mcan_config->custom;
+
+	return can_mcan_sys_write_mram(config->mram, offset, src, len);
+}
+
+static int can_mspm0_canfd_clear_mram(const struct device *dev, uint16_t offset, size_t len)
+{
+	const struct can_mcan_config *mcan_config = dev->config;
+	const struct can_mspm0_canfd_config *config = mcan_config->custom;
+
+	return can_mcan_sys_clear_mram(config->mram, offset, len);
+}
+
+static int can_mspm0_canfd_get_core_clock(const struct device *dev, uint32_t *rate)
+{
+	const struct can_mcan_config *mcan_config = dev->config;
+	const struct can_mspm0_canfd_config *config = mcan_config->custom;
+	const struct device *clk_dev = DEVICE_DT_GET(DT_NODELABEL(ckm));
+	uint32_t clock_rate;
+	int ret;
+
+	ret = clock_control_get_rate(clk_dev, (struct mspm0_sys_clock *)config->clock_subsys,
+				     &clock_rate);
+	if (ret < 0) {
+		return ret;
+	}
+
+	if (config->clock_cfg.divider != DL_MCAN_FCLK_DIV_1) {
+		*rate = clock_rate / (config->clock_cfg.divider >>  MCAN_CLKDIV_RATIO_OFS);
+	} else {
+		*rate = clock_rate;
+	}
+
+	return ret;
+}
+
+static int can_mspm0_canfd_clock_enable(const struct device *dev)
+{
+	const struct can_mcan_config *mcan_cfg = dev->config;
+	const struct can_mspm0_canfd_config *config = mcan_cfg->custom;
+	DL_MCAN_RevisionId revid_MCAN0;
+
+	DL_MCAN_setClockConfig((MCAN_Regs *)config->ti_canfd_base,
+			       (DL_MCAN_ClockConfig *)&config->clock_cfg);
+
+	/* Get MCANSS Revision ID. */
+	do {
+		DL_MCAN_enableModuleClock((MCAN_Regs *)config->ti_canfd_base);
+		DL_MCAN_getRevisionId((MCAN_Regs *)config->ti_canfd_base, &revid_MCAN0);
+	} while ((uint32_t)revid_MCAN0.scheme == 0x00);
```

Comment:
Could you please improve the comment here a bit? It's clear from the function names that this is fetching the revision ID, question is why this is needed...